File: lua/refactoring/tests/refactor/119/c/example/extract_var.start.c
#include <stdio.h>

#define max(a, b)                                                              \
  ({                                                                           \
    __typeof__(a) _a = (a);                                                    \
    __typeof__(b) _b = (b);                                                    \
    _a > _b ? _a : _b;                                                         \
  })

#define min(a, b)                                                              \
  ({                                                                           \
    __typeof__(a) _a = (a);                                                    \
    __typeof__(b) _b = (b);                                                    \
    _a < _b ? _a : _b;                                                         \
  })

struct Order {
  double quantity;
  double itemPrice;
};

double orderCalculation(struct Order order, int a, int b) {
  float temp = a * b;
  float blah = (order.quantity * order.itemPrice) - 7;

  printf("%f %f", temp, blah);

  return order.quantity * order.itemPrice -
         max(0.0, order.quantity - 500) * order.itemPrice * 0.05 +
         min(order.quantity * order.itemPrice * 0.1, 100.00);
}

File: lua/refactoring/tests/refactor/119/lua/global_scope/extract_var.start.lua
local a = 1
local b = 2
local order = {
    quantity = 3,
    itemPrice = 4,
}
local temp = a * b
local blah = (order.quantity * order.itemPrice) - 7

print(string.format("%d %d"), temp, blah)

return order.quantity * order.itemPrice
    - math.max(0, order.quantity - 500) * order.itemPrice * 0.05
    + math.min(order.quantity * order.itemPrice * 0.1, 100)
